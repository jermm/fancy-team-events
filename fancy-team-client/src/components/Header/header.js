import React, { Component } from 'react';
import EventLogo from "./calendar-events-svgrepo-com.svg";
import './header.scss';

function Header(props) {
    return (
        <div>
           <header className="event-header">
                <img src={EventLogo} className="event-logo" alt="Event header logo" width="200" height="300" />
                <h1 className='event-title'>Fancy Event Organizer</h1>
            </header> 
        </div>
    )
}

export default Header;