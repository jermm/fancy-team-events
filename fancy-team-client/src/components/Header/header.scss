.event-header {
    background-color: #282c34;
    min-height: 50vh;
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
  overflow-x: hidden;
  overflow-y: auto;
  font-family: "Quicksand", Arial, Helvetica, sans-serif;
  font-size: calc(10px + 2vmin);
    color: white;
    height:100%;
  }
  
  .event-logo {
    height: 30vmin;
    pointer-events: none;
  }