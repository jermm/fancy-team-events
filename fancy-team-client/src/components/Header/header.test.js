import React from 'react';
import ReactDOM from 'react-dom';
import Header from './header';

import { shallow } from 'enzyme';

it('Header renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<Header />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('Using shadown - should render correctly with no props', () => {
    const component = shallow(<Header />);
    expect(component).toMatchSnapshot();
  });
