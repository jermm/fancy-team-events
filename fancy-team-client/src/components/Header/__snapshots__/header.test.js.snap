// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Using shadown - should render correctly with no props 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <header
        className="event-header"
      >
        <img
          alt="Event header logo"
          className="event-logo"
          height="300"
          src="calendar-events-svgrepo-com.svg"
          width="200"
        />
        <h1
          className="event-title"
        >
          Fancy Event Organizer
        </h1>
      </header>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <img
            alt="Event header logo"
            className="event-logo"
            height="300"
            src="calendar-events-svgrepo-com.svg"
            width="200"
          />,
          <h1
            className="event-title"
          >
            Fancy Event Organizer
          </h1>,
        ],
        "className": "event-header",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "Event header logo",
            "className": "event-logo",
            "height": "300",
            "src": "calendar-events-svgrepo-com.svg",
            "width": "200",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Fancy Event Organizer",
            "className": "event-title",
          },
          "ref": null,
          "rendered": "Fancy Event Organizer",
          "type": "h1",
        },
      ],
      "type": "header",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <header
          className="event-header"
        >
          <img
            alt="Event header logo"
            className="event-logo"
            height="300"
            src="calendar-events-svgrepo-com.svg"
            width="200"
          />
          <h1
            className="event-title"
          >
            Fancy Event Organizer
          </h1>
        </header>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <img
              alt="Event header logo"
              className="event-logo"
              height="300"
              src="calendar-events-svgrepo-com.svg"
              width="200"
            />,
            <h1
              className="event-title"
            >
              Fancy Event Organizer
            </h1>,
          ],
          "className": "event-header",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "Event header logo",
              "className": "event-logo",
              "height": "300",
              "src": "calendar-events-svgrepo-com.svg",
              "width": "200",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Fancy Event Organizer",
              "className": "event-title",
            },
            "ref": null,
            "rendered": "Fancy Event Organizer",
            "type": "h1",
          },
        ],
        "type": "header",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
